src/api/oembedProxyApi.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (c) 2016-present, NDLA.
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import fetch from 'isomorphic-fetch';
+import queryString from 'query-string';
+import {
+  apiResourceUrl,
+  resolveJsonOrRejectWithError,
+  headerWithAccessToken,
+} from '../utils/apiHelpers';
+import { EmbedType } from '../interfaces';
+import { OembedEmbedType } from '../plugins/externalPlugin';
+
+export interface OembedProxyResponse {
+  type: string;
+  version: string;
+  title?: string;
+  description?: string;
+  authorName?: string;
+  authorUrl?: string;
+  providerName?: string;
+  providerUrl?: string;
+  cacheAge?: number;
+  thumbnailUrl?: string;
+  thumbnailWidth?: number;
+  thumbnailHeight?: number;
+  width?: number;
+  height?: number;
+  html?: string;
+}
+
+export const fetchOembed = async (
+  embed: EmbedType,
+  accessToken: string,
+): Promise<OembedEmbedType> => {
+  const url = new URL(typeof embed.data.url === 'string' ? embed.data.url : '');
+  if (url.hostname.includes('youtu') && url.protocol === 'http:') {
```

Comment:
Kunne nok egentlig gjort dette for dei fleste embeds. Ikkje sikker det er et stort problem lenger. Var et hack for å kunne vise gamle artikler med youtube-lenker etter at youtube slutta å godkjenne http-urler i oembed.

Reply:
Troor vi skal prøve å holde denne pr'en fri for endringer av funksjonalitet siden den er såpass stor.
Tar det evt i egen pr.

Reply:
Enig